specs/release.md

```diff
@@ -0,0 +1,319 @@
+# Release Specification
+
+Requirements for the continuous integration, continuous deployment, and release management processes that govern how code changes flow from development through production.
+
+---
+
+## Git Branching Strategy
+
+The repository MUST follow a trunk-based development model with short-lived feature branches merging to a protected main branch. This strategy prioritizes continuous integration and rapid feedback over long-running parallel development streams.
+
+### Main Branch
+
+The `main` branch represents the current state of truth and MUST always be in a deployable state. Every commit to `main` MUST pass all automated tests and type checking before merge. The branch MUST be protected with the following GitHub branch protection rules:
+
+- Pull request reviews REQUIRED before merge (minimum 1 approval for single-maintainer projects)
+- Status checks MUST pass before merge (tests, type checking, coverage thresholds)
+- Force pushes MUST be disabled
+- Deletion MUST be disabled
+- Linear history SHOULD be enforced (rebase or squash merge only)
+
+Direct commits to `main` are PROHIBITED except for emergency hotfixes when the pull request process would create unacceptable delay. Emergency commits MUST be documented with justification in the commit message.
+
+### Feature Branches
+
+All development work MUST occur on feature branches created from `main`. Feature branches SHOULD follow the naming convention `feature/{description}` or `fix/{description}` to indicate intent. Branch names MUST use kebab-case and SHOULD be concise but descriptive (e.g., `feature/oauth-pkce-support`, `fix/session-timeout-bug`).
+
+Feature branches MUST be short-lived, typically merged within 1-3 days of creation. Long-running feature branches increase merge conflicts and defer integration issues. For large features requiring longer development, use feature flags to merge incomplete work while keeping the functionality disabled in production.
+
+Feature branches SHOULD be deleted immediately after merge to reduce repository clutter and prevent accidental continued development on stale branches.
+
+### Hotfix Workflow
+
+Critical production issues requiring immediate deployment MAY bypass the standard pull request review process. Hotfixes MUST:
+
+1. Create a branch from `main` (not from a feature branch)
```

Comment:
I don't think this is quite right. Seen all feature branches branched off of main? I don't recall what we said we would do for production deploys and whether there would be a tracking branch but a hotfix to prod really needs to be done against the last prod commit so that invalidated features aren't accidentally deployed.